engine/runtime/src/test/java/org/enso/interpreter/test/NumbersTest.java

```diff
@@ -0,0 +1,94 @@
+package org.enso.interpreter.test;
+
+import java.util.Arrays;
+import java.util.Random;
+
+import org.enso.polyglot.MethodNames;
+import org.graalvm.polyglot.Context;
+import org.junit.AfterClass;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class NumbersTest extends TestBase {
+  private static final String[] OPERATIONS = {
+    " +", " -", " ^", " *", " %", " <=", " <", " >=", " >", " /",
+    ".div", ".bit_xor", ".bit_shift", ".bit_shift_r", ".bit_or", ".bit_and"
+  };
+
+  @Parameterized.Parameters(name="({1}){0} ({2})")
+  public static Object[][] parameters() {
+    var r = new Random();
+    var ops = Arrays.asList(OPERATIONS).stream().map(
+      (op) -> new Object[] { op, r.nextLong(), switch (op) {
+        case " ^" -> r.nextLong(10);
```

Comment:
It is a bit dangerous to let the right hand side argument of `^` to be `Long.MAX_VALUE` ;-)